api/v1alpha3/conditions_consts.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+// AzureCluster Conditions and Reasons
+const (
+	// NetworkInfrastructureReadyCondition reports of current status of cluster infrastructure
+	NetworkInfrastructureReadyCondition = "NetworkInfrastructureReady"
+	// APIServerProvisioningReason API Server endpoint for the loadbalancer
+	APIServerProvisioningReason = "APIServerProvisioning"
+	// APIServerProvisioningFailedReason used for failure during provisioning of API Server.
+	APIServerProvisioningFailedReason = "APIServerProvisioningFailed"
+)
+
+// AzureMachine Conditions and Reasons
+const (
+	// VMRunningCondition reports on current status of the Azure VM.
+	VMRunningCondition clusterv1.ConditionType = "VMRunning"
```

Comment:
Should these be replacing VMState? Or are they distinct and meant to stay distinct?

Reply:
I think they are meant to be distinct, at least for now while its a new change 

@fabriziopandini please correct me if I'm wrong

Reply:
I consider conditions are a tool to surface internal state, but then it up really up to you to decide how your internal state is managed and how many details you want to surface.
If you keep the two list separated, it can be verbose, but you get a certain degree of flexibility and decoupling